<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>NeuralNet</class>
 <widget class="QGroupBox" name="NeuralNet">
  <property name="enabled">
   <bool>true</bool>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>445</width>
    <height>430</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>NeuralNet</string>
  </property>
  <property name="title">
   <string>NeuralNet Calculations</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QPushButton" name="openTrainingButton">
       <property name="text">
        <string>Open Training</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="openTestButton">
       <property name="text">
        <string>Open Test</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QLabel" name="label">
       <property name="text">
        <string>Inputs:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="nrInputsSpinBox">
       <property name="minimum">
        <number>1</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_2">
       <property name="text">
        <string>Outputs:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="nroutputsLabel">
       <property name="text">
        <string>0</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label_3">
       <property name="text">
        <string>Layers:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="layersSpinBox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100000</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_4">
       <property name="text">
        <string>Neurons / layer:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="neuronsSpinBox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>100000</number>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_5">
       <property name="text">
        <string>Max iterations:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QSpinBox" name="maxIterSpinBox">
       <property name="minimum">
        <number>1</number>
       </property>
       <property name="maximum">
        <number>1000000000</number>
       </property>
       <property name="singleStep">
        <number>10000</number>
       </property>
       <property name="value">
        <number>100000</number>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_4">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <spacer name="horizontalSpacer_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_4">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label_9">
       <property name="text">
        <string>Connection rate:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="connectionRateSpinBox">
       <property name="minimum">
        <double>0.010000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_10">
       <property name="text">
        <string>Learning rate:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="learningRateSpinBox">
       <property name="minimum">
        <double>0.010000000000000</double>
       </property>
       <property name="maximum">
        <double>1.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.010000000000000</double>
       </property>
       <property name="value">
        <double>1.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_6">
       <property name="text">
        <string>Desired error:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QDoubleSpinBox" name="desiredErrorSpinBox">
       <property name="decimals">
        <number>5</number>
       </property>
       <property name="maximum">
        <double>10.000000000000000</double>
       </property>
       <property name="singleStep">
        <double>0.000100000000000</double>
       </property>
       <property name="value">
        <double>0.000000000000000</double>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_5">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <spacer name="horizontalSpacer_3">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="trainingDataTable">
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>70</number>
     </attribute>
     <attribute name="verticalHeaderDefaultSectionSize">
      <number>20</number>
     </attribute>
     <column>
      <property name="text">
       <string>New Column</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>New Column</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_2">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QPushButton" name="pushButton">
       <property name="text">
        <string>Train</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_3">
       <property name="text">
        <string>Save</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_4">
       <property name="text">
        <string>Load</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="pushButton_2">
       <property name="text">
        <string>Test</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QTableWidget" name="testDataTable">
     <attribute name="horizontalHeaderDefaultSectionSize">
      <number>70</number>
     </attribute>
     <attribute name="verticalHeaderDefaultSectionSize">
      <number>20</number>
     </attribute>
     <column>
      <property name="text">
       <string>New Column</string>
      </property>
     </column>
     <column>
      <property name="text">
       <string>New Column</string>
      </property>
     </column>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_5">
     <property name="topMargin">
      <number>0</number>
     </property>
     <item>
      <widget class="QLabel" name="label_7">
       <property name="text">
        <string>Min resolution:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="minResolutionEdit">
       <property name="enabled">
        <bool>true</bool>
       </property>
       <property name="text">
        <string>0.0001</string>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLabel" name="label_8">
       <property name="text">
        <string>Desired resolution:</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="desiredResolutionEdit"/>
     </item>
     <item>
      <widget class="QPushButton" name="matchButton">
       <property name="text">
        <string>Match</string>
       </property>
      </widget>
     </item>
     <item>
      <spacer name="horizontalSpacer_6">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>openTrainingButton</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>FileOpenTraining()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>43</y>
    </hint>
    <hint type="destinationlabel">
     <x>260</x>
     <y>2</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>nrInputsSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>NeuralNet</receiver>
   <slot>NrInputsChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>346</x>
     <y>44</y>
    </hint>
    <hint type="destinationlabel">
     <x>374</x>
     <y>16</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>Train()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>40</x>
     <y>241</y>
    </hint>
    <hint type="destinationlabel">
     <x>5</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_2</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>Test()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>407</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>319</x>
     <y>251</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>matchButton</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>Match()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>332</x>
     <y>409</y>
    </hint>
    <hint type="destinationlabel">
     <x>346</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_3</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>Save()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>251</y>
    </hint>
    <hint type="destinationlabel">
     <x>174</x>
     <y>264</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton_4</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>Load()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>252</x>
     <y>252</y>
    </hint>
    <hint type="destinationlabel">
     <x>256</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>openTestButton</sender>
   <signal>clicked()</signal>
   <receiver>NeuralNet</receiver>
   <slot>FileOpenTest()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>152</x>
     <y>39</y>
    </hint>
    <hint type="destinationlabel">
     <x>205</x>
     <y>17</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>NrInputsChanged(int)</slot>
  <slot>Train()</slot>
  <slot>Test()</slot>
  <slot>Match()</slot>
  <slot>Save()</slot>
  <slot>Load()</slot>
  <slot>FileOpenTest()</slot>
  <slot>FileOpenTraining()</slot>
 </slots>
</ui>
